specification/compiler/model/utils.ts

```diff
@@ -449,14 +450,9 @@ function setTags<TType extends model.BaseType | model.Property | model.EnumMembe
     }
   }
 
-  function getName (type: TType): string {
-    if (typeof type.name === 'string') {
-      return type.name
-    } else if (typeof type.name.name === 'string') {
-      return type.name.name
-    } else {
-      return 'unknown'
-    }
+  function getName (type): string {
+    if (type.name?.name != null) return type.name.name
+    return type.name
```

Comment:
Are we sure that `type.name` always exists?
```suggestion
    return type.name ?? 'unknown'
```

Reply:
Yes, `type` is `model.BaseType | model.Property | model.EnumMembe` that all have a `type.name` property that is either a `string` or a `TypeName`. Otherwise the TS compiler and linter would complain 😉